include/boost/integer/mod_inverse.hpp

```diff
@@ -33,12 +31,12 @@ boost::optional<Z> mod_inverse(Z a, Z modulus)
     if (a == 0)
     {
         // a doesn't have a modular multiplicative inverse:
-        return boost::none;
+        return 0;
```

Comment:
Does this make assumptions about a constructor for `Z` not being explicit? Or should this be `return Z();` or `return Z(0);` instead?

Reply:
Yeah good point. I also explicitly cast the comparisons to type `Z` so that we won't get warnings.